errorlint/lint.go

```diff
@@ -189,6 +189,27 @@ func LintErrorComparisons(fset *token.FileSet, info *TypesInfoExt) []Lint {
 			continue
 		}
 
+		hasErrorCase := false
+		for _, caseBlock := range switchStmt.Body.List {
+			caseClause, ok := caseBlock.(*ast.CaseClause)
+			if !ok {
+				continue
+			}
+			for _, clause := range caseClause.List {
+				switch clause := clause.(type) {
+				case nil:
+					continue // default label is safe
+				case *ast.Ident:
+					if clause.Name != "nil" {
+						hasErrorCase = true
```

Comment:
Could we return early after we find a nil clause?

Reply:
Opposite. Done as part of the refactor in [c1b11c1](https://github.com/polyfloyd/go-errorlint/pull/27/commits/c1b11c14ef4e4a6b6f079f3458e48b3398d7b359).